### PR TITLE
add _.clamp to README.md

### PR DESCRIPTION
**Pull request by tombiju**
- Adding lodash `_.clamp` function native replacement to `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2642,6 +2643,58 @@ Produces a random number between the inclusive lower and upper bounds. If only o
...
+
+_.clamp(10, -5);
+// => -5
+
+_.clamp(10, 99);
+// => 10
+
+// Native
+const clamp = (number, boundOne, boundTwo) => {
+  if (!boundTwo) {
```
Comment (tombiju) [inline README.md]: - The second argument is optional according to the docs: https://lodash.com/docs/4.17.15#clamp
-  With only two arguments, the function treats the first as the target number and the second as the upper bound
- With all three, the arguments are the target number, the lower bound, and then the upper bound
- So, if there are only two arguments, the number only has to be `clamped` against the upper bound

File: README.md [hunk 2]
```diff
@@ -2642,6 +2643,58 @@ Produces a random number between the inclusive lower and upper bounds. If only o
...
+    return Math.max(number, boundOne) === boundOne ? number : boundOne; 
+  } else if (Math.min(number, boundOne) === number) {
```
Comment (tombiju) [inline README.md]: - If the number is greater than the upper bound, the upper bound is return; likewise if the number is less than the lower bound, the lower bound is returned
- If nothing has been returned from those checks then that means the number does not need to be clamped as it falls within the bounds and the number itself is returned

File: README.md [hunk 3]
```diff
@@ -185,6 +185,7 @@ then Lodash/Underscore is the better option.*
 
 1. [_.inRange](#_inRange)
 2. [_.random](#_random)
+3. [_.clamp](#clamp)
```
Comment (cht8687) [inline README.md]: probably better to follow alphabetic order for the first charactor after . like how we order the names in other sections.

**Comment by cht8687** [comment]
Thanks for the contribution @tombiju 
Are you planning to add the rules and tests in the PR as well?

**Comment by tombiju** [comment]
@cht8687 I just added tests but not sure if I should/what to put for the rules as the native replacement is custom function as opposed to a one liner like `Object.values()` etc. I also saw that similar replacements such as the one for `inRange` don't have rules but have tests, so should I make a rule then even though those don't?

**Comment by cht8687** [comment]
@tombiju 

good point.  @cjtaylor1990 @stevemao  what do you think about this? (for alternatives which are functions and too long as description).

Could you address the alphabetic order comment if you can? thanks.

**Comment by cjtaylor1990** [comment]
@cht8687 For native alternatives that are too long to be practical, I feel like it may be worth mentioning them in the README file and then producing a warning that references the README for the linter rules. That way, the warning could still be short and concise, but then direct the user to where they could locate an alternative. While slightly inconvenient given that it's two steps instead of one step, I feel that it would be a bit easier to read for the user and make things more maintainable for us, especially as there is likely going to be longer alternatives that we want to add in the future.

This is somewhat related to the problem in Issue #177, where a general native solution would be far too cumbersome to even include in the README. I suggested mentioning that it can be possible to devise one and then giving an example for a specific use case.

**Comment by cht8687** [comment]
let's merge this and think about how to improve the rules for this kind of issue. 
thanks for the PR. @tombiju

**Pull request merged**